Dockerfile

```diff
@@ -8,19 +8,17 @@ FROM golang:1.15.10 as gobuilder
 # Build docker-gen from scratch
 FROM gobuilder as dockergen
 
-# Download the sources for the given version
 ARG DOCKER_GEN_VERSION
-ADD https://github.com/jwilder/docker-gen/archive/${DOCKER_GEN_VERSION}.tar.gz sources.tar.gz
 
-# Move the sources into the right directory
-RUN tar -xzf sources.tar.gz && \
-   mkdir -p /go/src/github.com/jwilder/ && \
-   mv docker-gen-* /go/src/github.com/jwilder/docker-gen
-
-# Install the dependencies and make the docker-gen executable
-WORKDIR /go/src/github.com/jwilder/docker-gen
-RUN go get -v ./... && \
-   CGO_ENABLED=0 GOOS=linux go build -ldflags "-X main.buildVersion=${DOCKER_GEN_VERSION}" ./cmd/docker-gen
+RUN git clone https://github.com/jwilder/docker-gen \
```

Comment:
:thinking: I find the need unclear to retrieve this repository completely.

Reply:
We're building a specific tag of docker-gen from sources using a multi stage build.

The sources are fetched then removed in a single layer in order to minimize the resultant intermediate image size.

The sources don't end up in the final image.

Reply:
* :thinking: I imagine that the cloning of such a repository will trigger transmission of unneeded data.
* Can the layer selection be influenced also by any other programming means according to the desire of a reduced image size?

Reply:
> 🤔 I imagine that the cloning of such a repository will trigger transmission of unneeded data.

Unneeded in regard to what ?

We want an image that can be built natively on any arch to be able to provide multi arch images straight from DockerHub. I think that building the required dependencies from sources and letting Docker buildx do the hard work rather than relying on pre built binaries that might or might not be available in the flavor we want / need is the better option here.

Methods to reduce the download size of a git clone with sparse checkout and shallow clone do exists, but they are very contrived and -from a maintainer point of view- aren't worth the hassle for a 4.72 Mib download.

> Can the layer selection be influenced also by any other programming means according to the desire of a reduced image size?

Let me clarify this: the fact that the sources of docker-gen are downloaded in the `dockergen` stage does not have any impact on the size of the final image your are pulling from DockerHub. During the build process, the builder just grab the required compiled docker-gen binary for inclusion in the final image and discard everything else from the `dockergen` stage.

Now regarding reducing the size of the image size, yes it might be possible but only marginally. The two biggest contributors to the image size are still the **docker-gen** and **forego** binaries.

Reply:
I suggest to take further design possibilities into account for the export and import of data which are really relevant from the referenced development repository.
Will tar files become applicable again?

Reply:
Could you elaborate on the problems you're experiencing with the approach of cloning **docker-gen**'s repository ? Maybe by opening a separate issue ?

Reply:
* I propose to avoid questionable data transmissions as much as possible.
* I imagine that relevant source files can be retrieved by selected tar files (instead of cloning the mentioned repository).